### ISSUE TITLE
Access NetGear received frame before decoding?

### ISSUE THREAD

**Issue by whogben** [abhiTronix/vidgear#142]
##  Question

Is it possible to access the NetGear frames before they're decoded or without decoding them?

NetGear is receiving frames in .jpg, and outputting them decoded for OpenCV use. Is it poss  

### Acknowledgment

- [X] I have searched the [issues](https://github.com/abhiTronix/vidgear/issues) for my issue and found nothing related or helpful.
- [X] I have read the [FAQs](https://abhitronix.github.io/vidgear/help/get_help/#frequently-asked-questions).
- [X] I have read the [Documentation](https://abhitronix.github.io/vidgear).

### Context

I am streaming to VidGear where I am using the decoded frames for OpenCV, and then encoding them as jpg with cv2.imencode. As the stream is already coming in to VidGear as .jpg, I was hoping to save the re-encoding by accessing it there.

**Comment by abhiTronix** [comment]
> I am streaming to VidGear where I am using the decoded frames for OpenCV, and then encoding them as jpg with cv2.imencode.

@whogben Are you directly implementing ZMQ server using pyzmq and OpenCV, not using VidGear's NetGear API?

> As the stream is already coming in to VidGear as .jpg, I was hoping to save the re-encoding by accessing it there.

**NetGear API never enables decoding at Client's end, until unless you specifically enabled Compression at Server's end too.** See following logic in NetGear API at Client's end, that governs if the received data _(from server)_ contains `compression` attribute or not:
https://github.com/abhiTronix/vidgear/blob/ba5de279e2299f73a79a05241fca6996e47395ad/vidgear/gears/netgear.py#L1018-L1029

**Comment by whogben** [comment]
Thanks!

I am setting up the server with compression:
```
options = {'flag' : 0, 'copy' : False, 'track' : False, 'compression_format': '.jpg', 'compression_param':[cv2.IMWRITE_JPEG_QUALITY, 85]}
self.server = NetGear(address = self.ip, port = self.port, protocol = 'tcp', pattern = 2, logging = True, **options)
```

And the client without explicit decompression:
```
options = {'flag' : 0, 'copy' : False, 'track' : False}
self.client = NetGear(address = addr, port = port, protocol = 'tcp',  pattern = 2, receive_mode = True, logging = True, **options)
```

I access the frames on the client with this:
```
frame = client.recv()
print('frame len ' + str(len(frame)))
print('frame: ' + str(frame))
```
And the output looks like this:
```
frame len 720
frame: [[[ 95 112 138]
  [ 95 112 138]
  [ 98 113 139]
  ...
  [136 135 139]
  [141 138 140]
  [146 143 145]]
... and so on, 2d array height x width x 3 channels
```

**Comment by abhiTronix** [comment]
@whogben So you want to just completely disable compression at  Client's end ? **I can implement that, please wait.**

**Comment by abhiTronix** [comment]
@whogben Btw, what are you going to do with `.jpg` data-frames at the client's end?

**Comment by abhiTronix** [comment]
@whogben Try `development` branch as follows:

```sh
  # clone the repository and get inside
  git clone https://github.com/abhiTronix/vidgear.git && cd vidgear

  # checkout the latest testing branch
  git checkout developement

  # install normally
  pip install .

  # OR install with asyncio support
  pip install .[asyncio]
```

**and if now you assign `'compression_format': None` at client's end, it will disable it. Try it and tell here if it works.**

**Comment by abhiTronix** [comment]
**@whogben  Closed and marked solved for now. Kindly share your results when you're ready. Good luck!**

**Comment by abhiTronix** [comment]
**Successfully resolved and merged in commit 52d7659**

**Comment by whogben** [comment]
Thank you @abhiTronix, confirming the new feature working for me in the current test branch - much appreciated!
